src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -321,11 +321,14 @@ public bool IsMemberAttributeContext(ISet<SyntaxKind> validTypeDeclarations, Can
             var token = this.TargetToken;
 
             if (token.Kind() == SyntaxKind.OpenBracketToken &&
-                token.Parent.IsKind(SyntaxKind.AttributeList) &&
-                this.SyntaxTree.IsMemberDeclarationContext(
-                    token.SpanStart, contextOpt: null, validModifiers: null, validTypeDeclarations: validTypeDeclarations, canBePartial: false, cancellationToken: cancellationToken))
+                token.Parent.IsKind(SyntaxKind.AttributeList))
             {
-                return true;
+                var isRecordParameter = token.Parent.IsParentKind(SyntaxKind.Parameter) &&
+                    token.Parent.Parent.IsParentKind(SyntaxKind.ParameterList) &&
+                    token.Parent.Parent.Parent.IsParentKind(SyntaxKind.RecordDeclaration);
```

Comment:
`if (token.Parent is { Parent: { Kind: Parameter, Parent: { Parent { Kind: RecordDeclaration } } } })`
?

Reply:
@CyrusNajmabadi I'll have to use RawKind for this form as there is no `Kind` property. The cast to int feels annoying to me. But let me know if this is okay for you.